mpt/impl/imap-mailbox/core/src/main/resources/org/apache/james/imap/scripts/ListWithSharedMailbox.test

```diff
@@ -23,50 +23,50 @@ C: a0 LIST "" ""
 S: \* LIST \(\\Noselect\) \"\.\" \"\"
 S: a0 OK LIST completed.
 
-C: a1 LIST "" "%"
```

Comment:
Imo you shoud reference the namespace of shared mailboxes even when the RFC-3501 LIST reference name is empty. Can you demonstrate this?

Can you also add user as part of the reference name?

Reply:
>Imo you shoud reference the namespace of shared mailboxes even when the RFC-3501 LIST reference name is empty. Can you demonstrate this?

I can't, the old code test should be fail (Listing.java). We can separate a ticket IMO

>Can you also add user as part of the reference name?

Yes, I can add more testing for that case

> And also a base mailbox name?

Sorry but I can't. I read the RFC, it said the second parameter is the mailbox name, not mailbox path. is that true?

Reply:
RFC says:

```
   Arguments:  reference name
               mailbox name with possible wildcards
```


        Imo you shoud reference the namespace of shared mailboxes even when the RFC-3501 LIST reference name is empty. Can you demonstrate this?

    I can't, the old code test should be fail (Listing.java). We can separate a ticket IMO

That's why I ask it. Note: PATH and USER should be here IMO only if it a delegated mailbox.

```
C: a1 LIST "" "%"
SUB {
S: \* LIST \(\\HasChildren\) \"\.\" \"#private.Boby.sharedMailbox\"
S: \* LIST \(\\HasNoChildren\) \"\.\" \"INBOX\"
}
S: a1 OK LIST completed.
```

Note: full path reference present only on shared mailbox. Relative path is enought for private mailboxes. (Thatn ensure we do not hurt current code ;-) )

```
C: a1 LIST "#private.Boby" "%"
SUB {
S: \* LIST \(\\HasChildren\) \"\.\" \"#private.Boby.sharedMailbox\"
}
S: a1 OK LIST completed.
```

Note: I can list mailboxes in Boby namespace.

```
C: a1 LIST "" "#private.Boby%"
```
For this last one I've got **no strong opinion**. I believe it should not return anything...